metsim/metsim.py

```diff
@@ -263,34 +263,41 @@ def launch(self):
         time_dim, groups = self._get_time_dim_and_time_dim()
 
         for label, times in groups.items():
-            self.pool = Pool(processes=nprocs)
-            self.site_generator = itertools.product(*iter_list)
+            self.site_generator = list(itertools.product(*iter_list))
```

Comment:
Is there a reason we need to compute this right away? Is it to use the `progress` function?

Reply:
Also, it might be worth renaming to `site_list` since it's no longer a generator.